### PR TITLE
app/vmctl: add support of basic auth and barer token

### PR DESCRIPTION
**Pull request by dmitryk-dk**
I added support for the bearer token for native protocol. 
I moved auth realization to its package. It helps to use it in other protocols if needed.
Made client initialization simpler

Related issue: https://github.com/VictoriaMetrics/VictoriaMetrics/issues/3835

### CODE REVIEW

**Comment by baconmania** [comment]
Now that we have the ability to set arbitrary headers via `vm-native-src-headers` / `vm-native-dst-headers`, do we need this? It's arguably weird to have special handling for specific headers as well as having the arbitrary header config for `vmctl`

**Comment by dmitryk-dk** [comment]
> Now that we have the ability to set arbitrary headers via `vm-native-src-headers` / `vm-native-dst-headers`, do we need this? It's arguably weird to have special handling for specific headers as well as having the arbitrary header config for

The flags was got from your realization. There could be different configurations of the source and destination db, so we should use different headers. 
Moving auth to own package simplifies parts of code, not just native protocol. I have added this just for native protocol here because of the issue (want to do one change for the single request). But in the future, this config can be used for different clients. And these changes will make it easy to add different auth realizations such as TLS etc.

File: app/vmctl/native/client.go [hunk 1]
```diff
@@ -18,11 +19,18 @@ const (
...
-	Headers     string
+}
+
+// New initialized an HTTP client
+func New(addr string, extraLabels []string, authCfg *auth.Config) *Client {
+	return &Client{
+		Addr:        addr,
+		ExtraLabels: extraLabels,
+		AuthCfg:     authCfg,
+	}
```
[the remark this answers is not in the document]
Comment (dmitryk-dk) [reply app/vmctl/native/client.go]: changed to struct usage

File: docs/CHANGELOG.md [hunk 2]
```diff
@@ -18,6 +18,7 @@ The following tip changes can be tested by building VictoriaMetrics components f
...
 * FEATURE: [vmctl](https://docs.victoriametrics.com/vmctl.html): add `--vm-native-src-headers` and `--vm-native-dst-headers` command-line flags, which can be used for setting custom HTTP headers during [vm-native migration mode](https://docs.victoriametrics.com/vmctl.html#native-protocol). Thanks to @baconmania for [the pull request](https://github.com/VictoriaMetrics/VictoriaMetrics/pull/3906).
+* FEATURE: [vmctl](https://docs.victoriametrics.com/vmctl.html): add `--vm-native-src-bearer-token` and `--vm-native-dat-bearer-token` command-line flags, which can be used for setting custom HTTP headers during [vm-native migration mode](https://docs.victoriametrics.com/vmctl.html#native-protocol). See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/3835).
```
[the remark this answers is not in the document]
Comment (dmitryk-dk) [reply docs/CHANGELOG.md]: fixed

**Review by hagen1778 — APPROVED** [review]
Thank you!

**Pull request merged**